pkg/kind/cluster.go

```diff
@@ -241,28 +246,61 @@ func (c *Cluster) ensureCorrectConfig(in []byte) (kindv1alpha4.Cluster, error) {
 	if err != nil {
 		return kindv1alpha4.Cluster{}, fmt.Errorf("parsing kind config: %w", err)
 	}
-
+	// the port and ingress-nginx label must be on the same node to ensure nginx runs on the node with the right port.
 	appendNecessaryPort := true
+	appendIngresNodeLabel := true
+	// pick the first node for the ingress-nginx if we need to configure node port.
+	nodePosition := 0
+
+	if parsedCluster.Nodes == nil || len(parsedCluster.Nodes) == 0 {
+		return kindv1alpha4.Cluster{}, fmt.Errorf("provided kind config does not have the node field defined")
+	}
+
 nodes:
 	for i := range parsedCluster.Nodes {
 		node := parsedCluster.Nodes[i]
 		for _, pm := range node.ExtraPortMappings {
 			if strconv.Itoa(int(pm.HostPort)) == c.cfg.Port {
 				appendNecessaryPort = false
+				nodePosition = i
+				if node.Labels != nil {
+					v, ok := node.Labels[ingressNginxNodeLabelKey]
+					if ok && v == ingressNginxNodeLabelValue {
+						appendIngresNodeLabel = false
+					}
+				}
+				break nodes
+			}
+		}
+		if node.Labels != nil {
+			v, ok := node.Labels[ingressNginxNodeLabelKey]
+			if ok && v == ingressNginxNodeLabelValue {
+				appendIngresNodeLabel = false
+				nodePosition = i
```

Comment:
I wonder if this can be further simplified. I am not fond of label breaks in go, but leave it up to you.

Reply:
It's really just comes down to last one winning or the first one winning. I chose first one winning. If we do last one wins, we can avoid using label breaks.

Reply:
i like first one winning too.

Here is an alternative way of writing it (thanks ChatGPT). I dont think it is any easier tho. let's leave it.

```
for i, node := range parsedCluster.Nodes {
    checkNodeLabels := func() {
        if node.Labels != nil {
            if v, ok := node.Labels[ingressNginxNodeLabelKey]; ok && v == ingressNginxNodeLabelValue {
                appendIngresNodeLabel = false
            }
        }
    }

    for _, pm := range node.ExtraPortMappings {
        if strconv.Itoa(int(pm.HostPort)) == c.cfg.Port {
            appendNecessaryPort = false
            nodePosition = i
            checkNodeLabels()
            break
        }
    }

    if appendNecessaryPort { // If the port was not matched, check labels
        checkNodeLabels()
        if !appendIngresNodeLabel {
            nodePosition = i
            break
        }
    }
}
```